### PR TITLE
Pinch gesture for osx (cleanup)

### PR DESCRIPTION
**Pull request by wrnrlr**
title says it all

### CODE REVIEW

File: io/pointer/pointer.go [hunk 1]
```diff
@@ -73,7 +77,7 @@ type PassOp struct {
 type ID uint16
 
 // Type of an Event.
-type Type uint8
+type Type uint16
```
Comment (eliasnaur) [inline io/pointer/pointer.go:80]: What motivates this change?

File: io/pointer/pointer.go [hunk 2]
```diff
@@ -36,6 +36,10 @@ type Event struct {
 	Position f32.Point
 	// Scroll is the scroll amount, if any.
 	Scroll f32.Point
+	// Scale is the rate of a pinch gesture
+	Magnification float32
+	// Rotation is the rate of a rotate gesture
+	Rotation float32
```
Comment (eliasnaur) [inline io/pointer/pointer.go:42]: I don't "rate" is right. According to the [documentation](https://developer.apple.com/documentation/appkit/nsevent/1526249-rotation) it seems rotation is a property of a tablet pen, and that it is in absolute degrees. If so, it seems more like a physical property (the angle of the pen) than a gesture (say, rotating a map with a two-finger gesture).

File: io/pointer/pointer.go [hunk 3]
```diff
@@ -36,6 +36,10 @@ type Event struct {
 	Position f32.Point
 	// Scroll is the scroll amount, if any.
 	Scroll f32.Point
+	// Scale is the rate of a pinch gesture
+	Magnification float32
```
Comment (eliasnaur) [inline io/pointer/pointer.go:40]: "Rate" doesn't sound right. Better to say

// Magnification is the change of scale from a pinch gesture.

**Review by eliasnaur — CHANGES_REQUESTED** [review]
I'm sceptical of this change. It seems better (albeit more work) to introduce gesture.Scale and gesture.Rotate similar to gesture.Drag. The advantages are portability, testability and better fit with Gio's pointer.Areas (for example, a pinch shouldn't register if the two fingers are in two different pointer.Areas).

**Comment by eliasnaur** [comment]
> I'm sceptical of this change. It seems better (albeit more work) to introduce gesture.Scale and gesture.Rotate similar to gesture.Drag. The advantages are portability, testability and better fit with Gio's pointer.Areas (for example, a pinch shouldn't register if the two fingers are in two different pointer.Areas).

Oh, and we don't have to add the new fields and types to pointer.Event, along with rules for routing the new event types.

File: io/pointer/pointer.go (hunk 1 above)
Comment (wrnrlr) [reply io/pointer/pointer.go:80]: With this change there are more then 8 pointer Types

**Comment by wrnrlr** [comment]
True it is a lot of work. The osx implementation if particularly confusing because then we need to handle NSEvent and NSTouch together. It is not clear to me how existing multitouch events like the two finger scroll and click should work in gesture, while also supporting those events on pointer.

**Comment by eliasnaur** [comment]
> True it is a lot of work. The osx implementation if particularly confusing because then we need to handle NSEvent and NSTouch together. It is not clear to me how existing multitouch events like the two finger scroll and click should work in gesture, while also supporting those events on pointer.

Two-finger scroll and (right?)-click cleanly map to events we have to handle from mice anyway, but it's true that it would be nice to have them in gesture instead of the low-level pointer interface. Pinch zoom and scroll, on the other hand, are more clearly gestures; they entail expanding pointer.Type and pointer.Event to support them, yet they're rarely used.

**Comment by wrnrlr** [comment]
ok we will do it like that

**Pull request closed without merging**